prow/config/jobs.go

```diff
@@ -47,9 +47,10 @@ func mergePreset(preset Preset, labels map[string]string, pod *v1.PodSpec) error
 	}
 	for _, e1 := range preset.Env {
 		for i := range pod.Containers {
-			for _, e2 := range pod.Containers[i].Env {
+			for idx, e2 := range pod.Containers[i].Env {
 				if e1.Name == e2.Name {
-					return fmt.Errorf("env var duplicated in pod spec: %s", e1.Name)
+					// remove e2 and use e1
+					pod.Containers[i].Env = append(pod.Containers[i].Env[:idx], pod.Containers[i].Env[idx+1:]...)
```

Comment:
Is it safe to modify a slice while iterating over it like this? I know this is okay for maps, but I didn't think it was okay for slices.

Also I think this is backwards. The value in the container should be the override of the value in the preset.